src/robusta/integrations/jira/client.py

```diff
@@ -131,6 +153,133 @@ def _get_default_project_id(self):
+        response = self._call_jira_api(url, http_method=HttpMethod.POST, json=payload) or {}
+        logging.debug(response)
+        issue_id = response.get("id")
+        if issue_id and issue_attachments:
+            self.add_attachment(issue_id, issue_attachments)
+
+    def update_issue(self, issue_id, issue_data):
+        summary = self._get_nested_property(issue_data, "summary", "")
+        description = self._get_nested_property(issue_data, "description", "")
+        endpoint = f"issue/{issue_id}"
+        url = self._get_full_jira_url(endpoint)
+
+        payload = {
+            "fields": {"summary": summary, "description": description},
+        }
+
+        logging.debug(f"Update issue '{issue_id}' with payload: {payload}")
+        response = self._call_jira_api(url, http_method=HttpMethod.PUT, json=payload) or {}
+        logging.debug(response)
+
+    def manage_issue(self, issue_data, alert_data, issue_attachments=None):
+        existing_issue = self._get_issue_for_labels(issue_data.get("labels", [])) or []
+        status = self._get_nested_property(alert_data, "status", -1)
+        source = self._get_nested_property(alert_data, "source", None)
+        alert_resolved = status == FindingStatus.RESOLVED
+        is_prometheus_alert = source == FindingSource.PROMETHEUS
+
+        if not is_prometheus_alert:
+            # It's not an alert fired from Prometheus, we simply create
+            # a Jira issue without other logic involved
+            self.create_issue(issue_data, issue_attachments)
+        elif existing_issue:
+            issue_done = self._check_issue_done(existing_issue)
+            issue_id = self._get_nested_property(existing_issue, "id", -1)
+            issue_resolution = self._get_nested_property(existing_issue, "fields.resolution.name", "")
+
+            if issue_done:
+                if alert_resolved:
+                    logging.info(
+                        f"Ignoring resolved alert that is already 'done' in Jira for issue with id '{issue_id}'"
+                    )
+                elif self.noReopenResolution and issue_resolution == self.noReopenResolution:
+                    logging.info(f"Jira issue '{issue_id}' has '{issue_resolution}' resolution, we won't re-open it")
+                elif self.reopenIssues:
+                    self.transition_issue(issue_id, self.reopenStatusName)
+                    self.update_issue(issue_id, issue_data)
+                else:
+                    self.create_issue(issue_data, issue_attachments)
+            else:
+                if alert_resolved:
+                    if self.sendResolved:
+                        self.transition_issue(issue_id, self.doneStatusName)
+                        self.comment_issue(issue_id, "Issue was marked as resolved by the Alertmanager")
+                    else:
+                        logging.warn("Alert is resolved but 'sendResolved' is false, so we don't update Jira")
```

Comment:
If we have a open issue and a _resolved_ alert arrives, and `sendResolved` is false, then we simply warn the user that we won't update Jira. After all, the user selected to NOT send resolved status to jira with the `senResolved` value.